cpp/src/arrow/flight/sql/client.h

```diff
@@ -329,6 +329,25 @@ class ARROW_FLIGHT_SQL_EXPORT FlightSqlClient {
   /// \param[in] info         The FlightInfo of the query to cancel.
   ::arrow::Result<CancelResult> CancelQuery(const FlightCallOptions& options,
                                             const FlightInfo& info);
+   
+  /// \brief Sets session options.
+  ///
+  /// \param[in] options            RPC-layer hints for this call.
+  /// \param[in] session_options    The session options to set.
+  ::arrow::Result<std::vector<SetSessionOptionResult>> SetSessionOptions(
```

Comment:
It may be good to make the session identifier explicit so that clients can use it across individual connections? Also to avoid implicit state.

Reply:
[see [above](https://github.com/apache/arrow/pull/34817#pullrequestreview-1366808831) for discussion]